frontend/components/Main/MainHeader.tsx

```diff
@@ -204,23 +206,42 @@ export const MainHeader = () => {
         18,
       ),
     );
+
     const olasRequiredToStake = Number(
       formatUnits(
         `${SERVICE_TEMPLATES[0].configuration.olas_required_to_stake}`,
         18,
       ),
     );
-    const monthlyGasEstimate = Number(
+
+    const requiredOlas = olasCostOfBond + olasRequiredToStake;
+
+    const requiredGas = Number(
       formatUnits(
         `${SERVICE_TEMPLATES[0].configuration.monthly_gas_estimate}`,
         18,
       ),
     );
 
-    if (
-      (totalOlasBalance ?? 0) < olasCostOfBond + olasRequiredToStake ||
-      (totalEthBalance ?? 0) < monthlyGasEstimate
-    ) {
+    const isDeployable = (() => {
+      // case where required values are undefined (not fetched from the server)
+      if (totalEthBalance === undefined) return false;
+      if (totalOlasBalance === undefined) return false;
+      if (!services) return false;
+
+      // deployment statuses where agent should not be deployed
+      // if (serviceStatus === DeploymentStatus.DEPLOYED) return false; // condition already checked above
```

Comment:
do we need this commented code line? if yes, don't understand how it's checked above

Reply:
the condition is covered on L179: https://github.com/valory-xyz/olas-operate-app/blob/d8d926b9d77efd7a3974443921f1fb23ffc1a0f0/frontend/components/Main/MainHeader.tsx#L179

had left the comment for context around why it's not included in the statuses where agent should not be deployable

can remove though, let me know 🙏